lib/orphans.nix

```diff
@@ -0,0 +1,104 @@
+{ lib }:
+let
+  inherit (lib)
+    types
+    optional
+    isDerivation
+    attrByPath
+    flatten
+    any
+    assertMsg
+  ;
+  inherit (builtins)
+    length
+    concatStringsSep
+    mapAttrs
+    attrValues
+    tryEval
+  ;
+  isAttr = types.attrs.check;
+in
+rec {
+  /* Return a list of maintainers from a package, empty list if inaccessible.
+
+     Example:
+       listMaintainers pkgs.foo
+       => [ { ... } ]
+       listMaintainers pkgs.bar
+       => [ ]
+       listMaintainers {}
+       => error: listMaintainers must receive a package
+  */
+  listMaintainers = p:
+  if isDerivation p then
+    (attrByPath [ "meta" "maintainers" ] [ ] p)
+  else
+    throw "listMaintainers must receive a package"
+  ;
+
+  /* Returns the amount of maintainers that a package have.
+
+     Example:
+       maintainersLength pkgs.foo
+       => 1
+       maintainersLength pkgs.bar
+       => 0
+       maintainersLength {}
+       => error: listMaintainers must receive a package
+  */
+  maintainersLength = p:
+    length (listMaintainers p);
```

Comment:
This doesn't seem like useful function. Please move it into the `let` before `rec`, so we don't have to worry about any call sites.
See also https://wiki.haskell.org/Fairbairn_threshold

EDIT: remove and compare `listMaintainers` result to empty list `[]` in the function below; move the function below to the let.

Reply:
The idea is to use `hasMaintainer` outside, firstly I've used that on `listOrphans` but I thought I could use that to list packages that don't have x maintainers for free.